tests/helpers/server.ts

```diff
@@ -0,0 +1,67 @@
+import { Locator, Page } from "@playwright/test";
+import { TIMEOUT } from "./constants";
+
+export type ServerSettingType = "checkbox" | "text" | "number" | "password";
+
+export const visitServerSettings = async (page: Page) => {
+  await page.goto("/ui/");
+  await page.getByRole("link", { name: "Settings", exact: true }).click();
+};
+
+export const activateSetting = async (page: Page, settingName: string) => {
+  await page
+    .locator("css=tr", { hasText: settingName })
+    .getByRole("button")
+    .click();
+};
+
+export const updateCheckbox = async (settingRow: Locator) => {
+  const checkbox = settingRow.locator(".p-checkbox__label");
+  await checkbox.click();
+};
+
+export const checkIfChangeButtonExist = async (settingRow: Locator) => {
+  const changeButton = settingRow.getByRole("button", {
+    name: "Change",
+    exact: true,
+  });
+  return changeButton.isVisible({ timeout: 1000 });
+};
+
+export const updateSetting = async (
+  page: Page,
+  settingName: string,
+  settingType: ServerSettingType,
+  content?: string,
+) => {
+  const settingRow = page.locator("css=tr", { hasText: settingName });
+  if (settingType === "checkbox") {
+    await updateCheckbox(settingRow);
+  } else {
+    const settingInput = settingRow.locator(`css=input[type=${settingType}]`);
+    if (content) {
+      await settingInput.fill(content);
+    }
+  }
+  await settingRow.getByRole("button", { name: "Save", exact: true }).click();
+  await page.waitForSelector("text=Setting updated.", TIMEOUT);
+};
+
+export const resetSetting = async (page: Page, settingName: string) => {
+  const settingRow = page.locator("css=tr", { hasText: settingName });
+  if (await checkIfChangeButtonExist(settingRow)) {
+    const removeButton = settingRow.getByRole("button", {
+      name: "Remove",
+      exact: true,
+    });
+    await removeButton.click();
+    await page.waitForSelector("text=Setting updated.", TIMEOUT);
+    return;
+  }
+
+  await settingRow
+    .getByRole("button", { name: "Reset to default", exact: true })
+    .click();
+  await settingRow.getByRole("button", { name: "Save", exact: true }).click();
+  await page.waitForSelector("text=Setting updated.", TIMEOUT);
```

Comment:
This notification should also be closed.

Shall we add more validation logic that the setting, to ensure the value is actually a default now? We could ensure the value is printed without bold font (as that distinguishes the non-default values from default values). Or validate against a given value provided to the reset function?

Reply:
this relates to the previous comment, I think validating against values is better since bold font is a design aspect that's more likely to change over time. I have added validations for resets.